src/pip/_internal/operations/prepare.py

```diff
@@ -91,17 +112,34 @@ def dist(self, finder):
             )
         return dist
 
-    def prep_for_dist(self):
-        self.req.run_egg_info()
-        self.req.assert_source_matches_version()
+    def prep_for_dist(self, finder):
+        build_requirements, isolate = self.req.get_pep_518_info()
```

Comment:
A comment here saying that this is for installing stuff into a PEP 518 environment would be nice. :)